contracts/RestrictedNORI.sol

```diff
@@ -0,0 +1,1089 @@
+    uint256 scheduleTrueTotal = schedule.totalClaimedAmount +
+      totalSupply(scheduleId);
+    // avoid division by 0
+    if (scheduleTrueTotal == 0) {
+      return 0;
+    }
+    uint256 balanceOfAccount = balanceOf(account, scheduleId);
+    uint256 claimedAmountForAccount = schedule.claimedAmountsByAddress[account];
+    uint256 revocableForAccount = ((claimedAmountForAccount +
+      balanceOfAccount) * (totalQuantityToRevoke)) / scheduleTrueTotal;
+
+    return revocableForAccount;
+  }
+
+  /**
+   * @dev The total amount of released tokens available at the current block timestamp for the schedule.
+   * Takes the maximum of either the calculated linearly released amount based on the schedule parameters,
+   * or the released amount floor, which is set at the current released amount whenever the balance of a
+   * schedule is decreased through revocation or withdrawal.
+   */
+  function _releasedBalanceOfSingleSchedule(uint256 scheduleId)
+    internal
+    view
+    returns (uint256)
+  {
+    Schedule storage schedule = _scheduleIdToScheduleStruct[scheduleId];
+    return
+      MathUpgradeable.max(
+        _linearReleaseAmountAvailable(scheduleId),
+        schedule.releasedAmountFloor
+      );
+  }
+
+  /**
+   * Grants `role` to `account`
+   *
+   * ### Requirements:
+   *    - SCHEDULE_CREATOR_ROLE cannot be granted to an owner of RestrictedNORI tokens
+   *
+   * [See OZ Access Control docs for more] (
+   *  https://docs.openzeppelin.com/contracts/3.x/access-control)
+   */
+  function _grantRole(bytes32 role, address account) internal virtual override {
+    if (role == SCHEDULE_CREATOR_ROLE) {
+      if (_addressToScheduleIdSet[account].length() > 0) {
+        revert RoleUnassignableToScheduleHolder({
+          account: account,
+          role: "SCHEDULE_CREATOR_ROLE"
+        });
+      }
+    }
+    super._grantRole(role, account);
+  }
+
+  // Disabled functions ===========================================
+  /**
+   * Overridden standard ERC1155.setApprovalForAll that will always revert
+   *
+   * @dev This function is not currently supported from external callers so we override it to revert.
+   */
+  function setApprovalForAll(address, bool) public pure override {
+    revert OperatorActionsNotSupported();
+  }
+
+  /**
+   * Overridden standard ERC1155.isApprovedForAll that will always revert
+   *
+   * @dev This function is not currently supported from external callers so we override it to revert.
+   */
+  function isApprovedForAll(address, address)
+    public
+    pure
+    override
+    returns (bool)
+  {
+    revert OperatorActionsNotSupported();
+  }
+
+  /**
+   * Overridden standard ERC1155.burn that will always revert
+   *
+   * @dev This function is not currently supported from external callers so we override it to revert.
+   */
+  function burn(
+    address,
+    uint256,
+    uint256
+  ) public pure override {
+    revert BurningNotSupported();
```

Comment:
might be worth instead just extracting what we need out of `ERC1155PresetMinterPauserUpgradeable` instead of overriding these explicitly. All of these could probably just instead be validated once in the `beforeTokenTransfer` hook

Reply:
might need more convo to understand this one

Reply:
ah ok I think I understand what this means now that you suggested not inheriting from `ERC1155PresetMinterPauserUpgradeable` (and not from `Burnable` either)... did exactly that, extracted out pausing directly and added to `RestrictedNORI`, changes are pushed

Reply:
You'll want to remove the unused errors now as well